### PR TITLE
Remove agent sources manager thread

### PR DESCRIPTION
**Pull request by Ozaq**
This is a draft PR to remove the AgentsSrcManager thread. I need some help to verify this is still working correctly. If so I can use this as a basis to do some further cleanup and open the PR proper.

### CODE REVIEW

**Comment by chraibi** [comment]
We have here a [demo](https://github.com/JuPedSim/jpscore/blob/master/demos/scenario_11_sources/inifile.xml) for sources.
I guess, a unit test would be a better help for you, right?

**Comment by Ozaq** [comment]
@chraibi I tried the demos and they run to completion, I wanted to make sure I am not overlooking something subtle. I would like to cleanup some of the internals of `AgentSrcManager` and `AgentsSource` while I am at it but before doing so I need to be sure its working in this state and then add a system test for it :)

**Comment by chraibi** [comment]
@Ozaq would you wait for #637 and #638?

**Comment by Ozaq** [comment]
@chraibi sure no problem

File: jpscore/src/main.cpp [hunk 1]
```diff
@@ -71,45 +71,36 @@ int main(int argc, char ** argv)
...
-        time(&endtime);
-
-        if(config.ShowStatistics()) {
-            sim.PrintStatistics(evacTime);
-        }
+    if(!sim.InitArgs()) {
+        LOG_ERROR("Could not start simulation. Check the log for prior errors");
+        return EXIT_FAILURE;
+    }
+    time_t starttime{};
```
Comment (zeroset) [inline jpscore/src/main.cpp]: Did you move InitArgs out of the time measuring on purpose?

File: libcore/src/Simulation.cpp [hunk 2]
```diff
@@ -420,7 +420,8 @@ void Simulation::RunHeader(long nPed)
     _iod->WriteFrame(firstframe, _building.get());
     //first initialisation needed by the linked-cells
     UpdateRoutesAndLocations();
-    ProcessAgentsQueue();
+    // KKZ: RunBody calls this as one of the firs things, hence this can be removed
+    // ProcessAgentsQueue();
```
Comment (zeroset) [inline libcore/src/Simulation.cpp]: should we keep this or remove this?

**Comment by Ozaq** [comment]
> Unfortunately it looks like the sources are not working anymore, or they do not spawn new agents at all. See output from `/demos/scenario_11_sources/inifile.xml`
> 
> ```
> [2020-01-16 14:02:37.589] [info] Simulation started with 0 pedestrians
> 
> [2020-01-16 14:02:49.244] [info] Simulation completed
> [2020-01-16 14:02:49.244] [info] Exec Time 12.00s
> [2020-01-16 14:02:49.244] [info] Evac Time 5000.00s
> [2020-01-16 14:02:49.244] [info] Realtime Factor 416.67x
> [2020-01-16 14:02:49.244] [info] Number of Threads 
> ```

Hmm looks like I made a mistake on the rebase, I’ll lok. into it

File: libcore/src/Simulation.cpp [hunk 3]
```diff
@@ -432,7 +432,9 @@ void Simulation::RunHeader(long nPed)
     _iod->WriteFrame(firstframe, _building.get());
     //first initialisation needed by the linked-cells
     UpdateRoutesAndLocations();
-    ProcessAgentsQueue();
+    // KKZ: RunBody calls this as one of the firs things, hence this can be removed
```
Comment (zeroset) [inline libcore/src/Simulation.cpp]: Since this is now called again the comment should be removed?!

File: libcore/src/pedestrian/AgentsSource.cpp [hunk 4]
```diff
@@ -51,7 +51,7 @@ AgentsSource::AgentsSource(
     _frequency(frequency),
     _maxAgents(max_agents),
     _groupID(group_id),
-    _caption(caption),
+    _caption(std::move(caption)),
```
Comment (zeroset) [inline libcore/src/pedestrian/AgentsSource.cpp]: Is this move for optimization purpose? Maybe not necessary at this point?!

**Comment by Ozaq** [comment]
Thx for taking care of the PR, still swamped with work :/

**Pull request merged**